src/events/attributes.rs

```diff
@@ -364,7 +364,7 @@ impl<'a> Iterator for Attributes<'a> {
             return None;
         }
 
-        let mut bytes = self.bytes.iter().enumerate().skip(self.position);
+        let mut bytes = (self.position..).zip(self.bytes[self.position..].iter());
```

Comment:
Skip straight to the section of the buffer we're interested in instead of iterating over the preceding bytes.

Reply:
I'm wonder: doesn't the iterator just jump to the right position for slices?

Reply:
It's been a few months but I remember isolating this as having a performance impact independent from the other change.